pytest_pyodide/hook.py

```diff
@@ -102,13 +102,13 @@ def pytest_pycollect_makemodule(module_path: Path, path: Any, parent: Any) -> No
 
 
 def pytest_generate_tests(metafunc: Any) -> None:
-    runtime = metafunc.config.option.runtime
+    if "runtime" in metafunc.fixturenames:
+        runtime = metafunc.config.option.runtime
 
-    if runtime == "all":
-        runtime = RUNTIMES
+        if runtime == "all":
+            runtime = RUNTIMES
 
-    if "runtime" in metafunc.fixturenames:
-        metafunc.parametrize("runtime", [runtime], scope="session")
+        metafunc.parametrize("runtime", [runtime], scope="module")
```

Comment:
I found that setting `scope="session"`actually makes tests very slow because it reorders the test.